pkg/controller/core/workload_controller.go

```diff
@@ -104,11 +143,28 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			return ctrl.Result{}, client.IgnoreNotFound(err)
 		}
 	case admitted:
+		var err error
+		var requeueAfter time.Duration
+		countingTowardsTimeout, recheckAfter := r.admittedNotReadyWorkload(&wl, realClock)
+		if countingTowardsTimeout {
+			if recheckAfter > 0 {
+				klog.V(4).InfoS("Workload not yet ready and did not exceed its timeout", "workload", req.NamespacedName.String(), "timeoutAfter", recheckAfter)
+				requeueAfter = recheckAfter
+			} else {
+				if err := workload.UpdateStatusIfChanged(ctx, r.client, &wl, kueue.WorkloadAdmitted, metav1.ConditionUnknown,
+					"CancellingAdmission", "Admission is being cancelled due to exceeded timeout"); err != nil {
+					return ctrl.Result{}, client.IgnoreNotFound(err)
+				}
+				klog.V(3).InfoS("Resetting admission of the workload due to exceeding the PodsReady timeout", "workload", req.NamespacedName.String())
```

Comment:
@ahg-g re: And would be consider a Workload with PodsReady=Unknown to be past the timeout?

This is the reason why: if the controller fails to patch the admission to nil, then in the next iteration the `admittedNotReadyWorkload` returns false, so the workload would never be considered for timeout again.

Reply:
Should we attempt to set the admission to nil and then set the WorkloadAdmitted status to unknown?

Reply:
Why not use an explicit conditionType like `workloadTimeout`, and set the condition to True, `Unknown` is meaningless here. Then we can have another case and patch the admission to nil.

Reply:
> Should we attempt to set the admission to nil and then set the WorkloadAdmitted status to unknown?

It is tricky, if we set `admission=nil`, then scheduler will re-admit it immediately and it will be suspected to timeout validation again. This would trigger a cascade of re-admissions.

Reply:
> Why not use an explicit conditionType like `workloadTimeout`, and set the condition to True, `Unknown` is meaningless here. Then we can have another case and patch the admission to nil.

The situation when we cancel admission is very transient and it involves modifying two fields, which cannot be updated in the same request: `status.conditions` and `spec.admission`. So, I thought setting to `Unknown` makes sense to reflect that it is a transient state. Setting it directly to `False` would work equally well too. However, setting it to `False` directly will work as well.

The only issue with this (regardless if we set it to `Unknown` or `False`) is that in case of an error between setting the condition and setting `admission=nil` the timeout starts over. I guess we could ticket it as a separate issue, which will be easy to solve if the `admission` field lives in the `status`, as proposed here: https://github.com/kubernetes-sigs/kueue/issues/23.

Reply:
That being said, for now I added code to move the `Admitted=Unknown` to `Admitted=False` when `admission=nil`. Now the `Unknown` is transient - once `admitted=nil` it is set to `False`, but if there is an error, then the workload is reconciled again and it is put back to `True`. The timeout will restart counting.

Reply:
We still didn't address Aldo's scenario, right? Should we attempt to set `admission=nil` when admission is not nil and status is unknown?

Reply:
> We still didn't address Aldo's scenario, right?

Yes, if there is an error in the current implementation between setting the condition to `Unknown` and `admission` to nil, then the timeout will count again. 

I discussed this today with @alculquicondor who suggested to first try setting `admission=nil` and then `Admitted=False`. Then, in the event handlers delay requeing after both `admission` and the condition are updated. I'm going to test this approach.

Reply:
@ahg-g @alculquicondor I have implemented the approach with setting the `admission=nil` and it appears to work. I looped the integration tests 30 times and all passed. Also tested manually. 

This solves the problem of an error between setting the condition and admission. Now, once the admission is set to `nil` workload_controller recognizes the status as `cancellingAdmission` and it will retry to update the `Admitted` condition to `False `eventually. Also, scheduler will not attempt immediate re-admission, because it would requeue the workload only if its admission is cancelled fully. PTAL